ui/admin/app/controllers/scopes/scope/session-recordings/session-recording/channels-by-connection/index.js

```diff
@@ -33,4 +44,42 @@ export default class ScopesScopeSessionRecordingsSessionRecordingChannelsByConne
       this.model?.sessionRecording?.state === STATE_SESSION_RECORDING_UNKNOWN
     );
   }
+
+  // =actions
+
+  /**
+   * Reapplies storage policy dates to session recording
+   * @param {SessionRecordingModel}
+   */
+  @action
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess(({ end_time }) =>
+    !end_time
+      ? 'resources.policy.messages.later'
+      : 'resources.policy.messages.reapply',
+  )
+  async reapplyStoragePolicy(sessionRecording) {
```

Comment:
non-blocking: Is there benefit to passing in the sessionRecording versus using `this.model.sessionRecording` directly?
Same for delete.

Reply:
yup, so in a lot of cases the actions for crudl are injected in other controllers and reused in different routes as a result we have a pattern of passing in the model as a parameter. In this case these actions are only used in one route however its best conserve that pattern and it makes refactoring easier.

Reply:
Well said. Thinking bigger picture I agree with your statement.